### PR TITLE
Added new feature enhancements and fixed orientation issues

### PR DESCRIPTION
**Pull request by Adw41t**
Fixed open issues 
#9  : Fixed orientation issue and location in window
#23 : Added Skip for sequence of ShowCaseView
#22 : Added feature to change color of Pointer
#36 : Added feature to change color of Message text, Message Box, and line connecting Message Box and Pointer

### CODE REVIEW

File: showcaseviewlib/src/main/java/smartdevelop/ir/eram/showcaseviewlib/GuideView.java [hunk 1]
```diff
@@ -88,6 +95,11 @@
     private DismissType dismissType;
     private PointerType pointerType;
     private final GuideMessageView mMessageView;
+    private final TextView skipButton;
+    private View lastTargetView;
+    //Skip button is false by default
+    private boolean setSkip =false;
```
Comment (mreram) [inline showcaseviewlib/src/main/java/smartdevelop/ir/eram/showcaseviewlib/GuideView.java:101]: Reformat, i prefer enableSkipButton and remove your comment

File: showcaseviewlib/src/main/java/smartdevelop/ir/eram/showcaseviewlib/GuideView.java [hunk 2]
```diff
@@ -111,13 +123,27 @@ private GuideView(Context context, View view) {
         }
 
         mMessageView = new GuideMessageView(getContext());
+        skipButton = new TextView(context);
+        skipButton.setText("SKIP");
```
Comment (mreram) [inline showcaseviewlib/src/main/java/smartdevelop/ir/eram/showcaseviewlib/GuideView.java:127]: Remove this hard code and set a variable it doesn't support multi-language and a developer can't change this.

File: showcaseviewlib/src/main/java/smartdevelop/ir/eram/showcaseviewlib/GuideView.java [hunk 3]
```diff
@@ -111,13 +123,27 @@ private GuideView(Context context, View view) {
...
+        skipButton.setTextColor(Color.WHITE);
+        skipButton.setGravity(android.view.Gravity.CENTER);
+        skipButton.setPadding(
+                messageViewPadding,
+                messageViewPadding,
+                messageViewPadding,
+                messageViewPadding
+        );
+        skipParams = new FrameLayout.LayoutParams(FrameLayout.LayoutParams.WRAP_CONTENT, FrameLayout.LayoutParams.WRAP_CONTENT);
+        ((LayoutParams)skipParams).setMargins(0,0,10,140);
```
Comment (mreram) [inline showcaseviewlib/src/main/java/smartdevelop/ir/eram/showcaseviewlib/GuideView.java:137]: Remove these magic numbers! why are you using 10 and 140?! they are so weird.

Comment (mreram) [reply showcaseviewlib/src/main/java/smartdevelop/ir/eram/showcaseviewlib/GuideView.java:137]: Also you must convert dp to pixel

File: showcaseviewlib/src/main/java/smartdevelop/ir/eram/showcaseviewlib/GuideView.java [hunk 4]
```diff
@@ -126,7 +152,13 @@ private GuideView(Context context, View view) {
                 ViewGroup.LayoutParams.WRAP_CONTENT
             )
         );
-
+        skipButton.setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if(setSkip && lastTargetView!=null)
```
Comment (mreram) [inline showcaseviewlib/src/main/java/smartdevelop/ir/eram/showcaseviewlib/GuideView.java:158]: Reformat

File: showcaseviewlib/src/main/java/smartdevelop/ir/eram/showcaseviewlib/GuideView.java [hunk 5]
```diff
@@ -248,9 +280,23 @@ private boolean isLandscape() {
         return display_mode != Configuration.ORIENTATION_PORTRAIT;
     }
 
+    private int checkOrientation(){
+        try{
+            Display display = ((WindowManager) getContext().getSystemService(Context.WINDOW_SERVICE)).getDefaultDisplay();
```
Comment (mreram) [inline showcaseviewlib/src/main/java/smartdevelop/ir/eram/showcaseviewlib/GuideView.java:285]: Reformat

File: showcaseviewlib/src/main/java/smartdevelop/ir/eram/showcaseviewlib/GuideView.java [hunk 6]
```diff
@@ -438,6 +484,27 @@ public void show() {
...
+                ((LayoutParams)skipParams).gravity = android.view.Gravity.RIGHT | android.view.Gravity.BOTTOM;
+                break;
+
+            case Surface.ROTATION_90:
+                ((LayoutParams)skipParams).setMargins(10,0,0,0);
+                ((LayoutParams)skipParams).gravity = android.view.Gravity.LEFT | android.view.Gravity.BOTTOM;
+                break;
+
+            case Surface.ROTATION_270:
+                ((LayoutParams)skipParams).setMargins(0,0,10,0);
```
Comment (mreram) [inline showcaseviewlib/src/main/java/smartdevelop/ir/eram/showcaseviewlib/GuideView.java:501]: You have a lot of magic numbers

**Comment by mreram** [comment]
Hi @Adw41t, it's an interesting change but this change has a lot of issues please fix them and notify me.

File: showcaseviewlib/src/main/java/smartdevelop/ir/eram/showcaseviewlib/GuideView.java (hunk 5 above)
Comment (mreram) [reply showcaseviewlib/src/main/java/smartdevelop/ir/eram/showcaseviewlib/GuideView.java:285]: It's more than 100character in one line please use this format or something like that:
```
Display display = ((WindowManager) getContext().getSystemService(
      Context.WINDOW_SERVICE
)).getDefaultDisplay();
```

File: showcaseviewlib/src/main/java/smartdevelop/ir/eram/showcaseviewlib/GuideView.java [hunk 7]
```diff
@@ -438,6 +484,27 @@ public void show() {
...
+        if(setSkip) {
+
+        switch(checkOrientation()){
+            case Surface.ROTATION_0:
+                ((LayoutParams)skipParams).setMargins(0,0,10,140);
+                ((LayoutParams)skipParams).gravity = android.view.Gravity.RIGHT | android.view.Gravity.BOTTOM;
+                break;
+
+            case Surface.ROTATION_90:
+                ((LayoutParams)skipParams).setMargins(10,0,0,0);
```
Comment (mreram) [inline showcaseviewlib/src/main/java/smartdevelop/ir/eram/showcaseviewlib/GuideView.java:496]: Magic numbers

File: showcaseviewlib/src/main/java/smartdevelop/ir/eram/showcaseviewlib/GuideView.java [hunk 8]
```diff
@@ -438,6 +484,27 @@ public void show() {
         startAnimation.setFillAfter(true);
         this.startAnimation(startAnimation);
         mIsShowing = true;
+        if(setSkip) {
+
+        switch(checkOrientation()){
+            case Surface.ROTATION_0:
+                ((LayoutParams)skipParams).setMargins(0,0,10,140);
```
Comment (mreram) [inline showcaseviewlib/src/main/java/smartdevelop/ir/eram/showcaseviewlib/GuideView.java:491]: Magic numbers

**Comment by Adw41t** [comment]
Hi @mreram, resolved issues as per your suggestions. Thank you for reviewing and guiding!!

**Pull request closed without merging**